src/components/map/CountriesTileLayer.js

```diff
@@ -0,0 +1,97 @@
+import React, { useEffect, useState } from "react";
+import { useMap } from "react-leaflet";
+import L from 'leaflet';
+import { layerStyle, highlightStyles } from './MapStyle';
+import _ from "lodash";
+import { createAltPopup } from "../../utils/mapUtils"
+import ReactDOMServer from 'react-dom/server';
+// eslint-disable-next-line no-unused-vars
+import VectorGrid from "leaflet.vectorgrid";
+
+const toggleCountrySelection = (e, layer, highlightStyle) => {
+  // only change selection style on countries that have seroprev data
+  const layerStyle = layer._overriddenStyles[e.layer.properties.CODE]; //TODO: refactor this kinda ugly
+
+  if (layerStyle) 
+    layer.setFeatureStyle(e.layer.properties.CODE, highlightStyle);
+}
+
+export default function CountriesTileLayer(props) {
+
+  const { url, records, zIndex } = props;
+  const map = useMap();
+  const [layer, setLayer] = useState(undefined);
+  
+  useEffect(() => {
+        
+    var layer = L.vectorGrid.protobuf(`${url}/tile/{z}/{y}/{x}.pbf`, {
+      rendererFactory: L.svg.tile,
+      attribution: '',
+      interactive: true,
+      vectorTileLayerStyles: layerStyle,
+      minZoom: 1,
+      minNativeZoom: 3,
+      getFeatureId: (feature) => feature.properties.CODE,
+      zIndex: zIndex
+    }
+    ).addTo(map);
+
+    setLayer(layer);
+  },[])
+
+  useEffect(() => {
+    if ( layer )
+    {
+      records.forEach(element => {
+        if (element.properties.testsAdministered != null)
```

Comment:
Consider using element.hasOwnProperty('testsAdministered') to avoid searching through the whole prototype chain

https://stackoverflow.com/questions/455338/how-do-i-check-if-an-object-has-a-key-in-javascript